frame/alliance/src/lib.rs

```diff
@@ -606,25 +624,56 @@ pub mod pallet {
 		}
 
 		/// Initialize the founders, fellows, and allies.
+		/// Resets members and removes all active proposals if alliance is already initialized.
 		///
-		/// This should only be called once, and must be called by the Root origin.
-		#[pallet::weight(T::WeightInfo::init_members(
+		/// Must be called by the Root origin.
+		/// To force reset the Alliance, witness data must be provided.
+		#[pallet::weight(T::WeightInfo::force_set_members(
 			T::MaxFounders::get(),
 			T::MaxFellows::get(),
-			T::MaxAllies::get()
+			T::MaxAllies::get(),
+			witness.proposals,
+			witness.members,
 		))]
-		pub fn init_members(
+		pub fn force_set_members(
 			origin: OriginFor<T>,
 			founders: Vec<T::AccountId>,
 			fellows: Vec<T::AccountId>,
 			allies: Vec<T::AccountId>,
+			witness: ForceSetWitness,
 		) -> DispatchResult {
 			ensure_root(origin)?;
 
-			// Cannot be called if the Alliance already has Founders or Fellows.
-			// TODO: Remove check and allow Root to set members at any time.
-			// https://github.com/paritytech/substrate/issues/11928
-			ensure!(!Self::is_initialized(), Error::<T, I>::AllianceAlreadyInitialized);
+			if Self::is_initialized() {
+				// Reset Alliance by removing all members.
+				// Veto and remove all active proposals to avoid any unexpected behavior from
+				// actionable items managed outside of the pallet. Items managed within the pallet,
+				// like `UnscrupulousWebsites`, are left for the new Alliance to clean up or keep.
+
+				ensure!(
+					T::ProposalProvider::proposals_count() <= witness.proposals,
+					Error::<T, I>::BadWitness
+				);
+				ensure!(
+					Self::votable_members_count() <= witness.members,
+					Error::<T, I>::BadWitness
+				);
+
+				T::ProposalProvider::proposals().into_iter().for_each(|hash| {
+					T::ProposalProvider::veto_proposal(hash);
+				});
+
+				T::MembershipChanged::change_members_sorted(&[], &Self::votable_members(), &[]);
+				ensure!(
+					// `MemberRole` variants count never greater than members count.
+					Members::<T, I>::clear(witness.members, None).maybe_cursor.is_none(),
```

Comment:
Maybe there are non-votable members in here as well? Then the cursor is not `None`.

Reply:
Yes, changed it to check agains `member_count` not `voting_member_count`.
Trying to reproduce locally

Reply:
in unit tests never return me `not none cursor`, probably different impl of clear

Reply:
fixed now